### PR TITLE
Add online visualization of spirit based on state estimate topic

### PR DESCRIPTION
**Pull request by jcnorby**

### CODE REVIEW

File: spirit_utils/src/robot_interface.cpp [hunk 1]
```diff
@@ -18,14 +18,12 @@ RobotInterface::RobotInterface(ros::NodeHandle nh) {
...
 void RobotInterface::controlInputCallback(const spirit_msgs::ControlInput::ConstPtr& msg) {
-  // ROS_INFO("In controlInputCallback");
 }
 
 void RobotInterface::publishJointEncoders() {
-  // ROS_INFO("In publishJointEncoders");
   sensor_msgs::JointState msg;
 
-  std::vector<std::string> joint_names = {"0","1","2","3","4","5","6","7","8","9","10","11"};
+  std::vector<std::string> joint_names = {"j8", "j0", "j1", "j9","j2", "j3", "j10", "j4","j5", "j11", "j6", "j7"};
```
Comment (nikolaif399) [inline spirit_utils/src/robot_interface.cpp:26]: These don't need to align with the URDF specified joint names do they? Those should still be 0-11.

File: spirit_utils/src/rviz_interface.cpp [hunk 2]
```diff
@@ -123,6 +127,29 @@ void RVizInterface::footstepPlanCallback(const spirit_msgs::FootstepPlan::ConstP
...
+void RVizInterface::stateEstimateCallback(const spirit_msgs::StateEstimate::ConstPtr& msg) {
+
+  // Make a transform message for the body, populate with state estimate data, and publish
+  geometry_msgs::TransformStamped transformStamped;
+  transformStamped.header = msg->header;
+  transformStamped.header.frame_id = map_frame_;
+  transformStamped.child_frame_id = "dummy";
+  transformStamped.transform.translation.x = msg->body.pose.pose.position.x;
+  transformStamped.transform.translation.y = msg->body.pose.pose.position.y;
+  transformStamped.transform.translation.z = msg->body.pose.pose.position.z;
```
Comment (nikolaif399) [inline spirit_utils/src/rviz_interface.cpp:139]: Irritating that geometry_msgs::Transform and geometry_msgs::Pose use different representations of a 3D point, not worth doing anything about it though.

**Review by nikolaif399 — APPROVED** [review]
Only comment worth reviewing is the change in joint names for '0','1',... to 'j0','j1',...

File: spirit_utils/src/robot_interface.cpp (hunk 1 above)
Comment (jcnorby) [reply spirit_utils/src/robot_interface.cpp:26]: Actually that's worth discussing - I used the urdf with the stl meshes which uses j0 j1 ... for the joint names. We should probably make sure we're consistent about that naming scheme.

**Comment by nikolaif399** [comment]
Going to pull this is now. URDF joint naming discussed on slack.

**Pull request merged**